cli/src/cmd_update.rs

```diff
@@ -0,0 +1,73 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2025 Oxide Computer Company
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use bytes::Bytes;
+use clap::Parser;
+use futures::StreamExt;
+use oxide::{Client, ClientExperimentalExt};
+use tokio::fs::File;
+use tokio_util::io::ReaderStream;
+
+use crate::{generated_cli::CliConfig, AuthenticatedCmd, OxideOverride};
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+pub struct CmdUpload {
+    /// Path to the TUF repository
+    #[clap(short, long)]
+    path: PathBuf,
+
+    /// Name for the repository (defaults to final path component of --path)
+    #[clap(long)]
+    filename: Option<String>,
```

Comment:
I don't understand the use-case for this flag.

Is it so that the user can upload multiple copies of the same TUF repo with different files names? I feel like it's more likely to confuse the user than offer utility.